components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1131,11 +1146,37 @@ private class MonitoredItemWrapper {
                             EventTemplate.PendingAlarms.RetainIndex = eventFilter.SelectClauses.Count();
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
+
+                        // set up the timers
+                        if (EventTemplate.PendingAlarms.UpdateIntervalTimespan.HasValue) {
+                            _pendingAlarmsUpdateTimer.Interval = EventTemplate.PendingAlarms.UpdateIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsUpdateTimer.Elapsed += OnPendingAlarmsUpdateTimerElapsed;
+                            _pendingAlarmsUpdateTimer.Start();
+                        }
+                        if (EventTemplate.PendingAlarms.SnapshotIntervalTimespan.HasValue) {
+                            _pendingAlarmsSnapshotTimer.Interval = EventTemplate.PendingAlarms.SnapshotIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsSnapshotTimer.Elapsed += OnPendingAlarmsSnapshotTimerElapsed;
+                            _pendingAlarmsSnapshotTimer.Start();
+                        }
                     }
                     Item.Filter = eventFilter;
                 }
             }
 
+            private void OnPendingAlarmsUpdateTimerElapsed(object sender, System.Timers.ElapsedEventArgs e) {
+                if (EventTemplate?.PendingAlarms?.Dirty == true) {
+                    _pendingAlarmsUpdateTimer.Enabled = false;
+                    SendPendingAlarms(false);
+                    _pendingAlarmsUpdateTimer.Enabled = true;
+                }
+            }
+
+            private void OnPendingAlarmsSnapshotTimerElapsed(object sender, System.Timers.ElapsedEventArgs e) {
+                _pendingAlarmsSnapshotTimer.Enabled = false;
```

Comment:
is this supposed to synchronize? If yes I don't think its 100% safe. Is it possible to not use autoreset and rearm here instead on exit?

Reply:
Autoreset is now off so we shouldn't have any sync issues. Good point.